packages/core/lib/loader.js

```diff
@@ -1,70 +1,90 @@
 'use strict';
 
-const { dirname, join } = require('path');
+const { basename, dirname, join } = require('path');
 
 const cosmiconfig = require('cosmiconfig');
 
-const { resolve, resolvePreset, isResolveError } = require('./resolve');
+const { load: loadEnv } = require('dotenv');
+const { sync: findUp } = require('find-up');
+
 const { merge } = require('./utils');
+const { createResolver } = require('./resolver');
 
 exports.createLoader = (namespace) => {
-  const loadConfig = (context, config) => {
-    const { loadSync, searchSync } = cosmiconfig(namespace, {
-      stopDir: context,
-    });
-    return config
-      ? loadSync(resolvePreset(context, config))
-      : searchSync(context);
+  const { resolve, resolvePreset, isResolveError } = createResolver();
+
+  const loadOrSearchConfig = (stopDir, module) => {
```

Comment:
I find that I have to reread this function multiple times from its different call sites - it seems a bit overloaded too. Maybe it makes sense to split it into two separate functions which then have clearly separated responsibilities?